### PR TITLE
[sw,tests] Remove the duplicate kmac smoke test

### PR DESCRIPTION
**Pull request by doreis-lowrisc**
This PR only removes update the `sw/device/tests/kmac_smoketest.c` and delete the `sw/device/tests/dif/dif_kmac_smoketest.c` as pointed out by #9363

### CODE REVIEW

File: sw/device/tests/kmac_smoketest.c [hunk 1]
```diff
@@ -151,10 +150,62 @@ const shake_test_t shake_tests[] = {
...
                    0x6a82fcf4, 0xa91b0815},
-        .digest_len = 102,  // Rate (r) is 34 words.
+        .digest_len = 3264 / 8 / 4,  // Rate (r) is 34 words.
     },
 };
 
+/**
+ * Run a SHA-3 test case with varying alignments.
+ */
+void run_sha3_alignment_test(dif_kmac_t *kmac) {
```
Comment (mundaym) [inline sw/device/tests/kmac_smoketest.c:160]: We should either inline this OR move the other two loops into their own functions. Right now it is a bit inconsistent that this is the only test that has been pulled out of line.

Comment (mundaym) [reply sw/device/tests/kmac_smoketest.c:160]: Probably makes sense to pull everything out of line now that there are more tests.

**Review by timothytrippel — APPROVED** [review]
Thanks @doreis-lowrisc !

File: sw/device/tests/kmac_smoketest.c [hunk 2]
```diff
@@ -109,6 +107,8 @@ typedef struct shake_test {
 /**
  * SHAKE tests.
  */
+// Examples generated using a custom Go program importing library
+// `lang.org/x/crypto/sha3`
```
Comment (mundaym) [inline sw/device/tests/kmac_smoketest.c:111]: nit: `golang.org/x/crypto/sha3`

File: sw/device/tests/kmac_smoketest.c [hunk 3]
```diff
@@ -109,6 +107,8 @@ typedef struct shake_test {
 /**
  * SHAKE tests.
  */
+// Examples generated using a custom Go program importing library
```
Comment (mundaym) [inline sw/device/tests/kmac_smoketest.c:110]: teeny tiny nit: `s/library/package/`

**Review by mundaym — APPROVED** [review]
Some small nits to address and then I think this can be merged. Thanks @doreis-lowrisc.

**Pull request merged**